### PR TITLE
Various gpt-index fixes

### PR DESCRIPTION
**Pull request by Hikari-Haru**
Fix regression in full discord server indexing
Add back response mode, only on vector index queries
Added the llmpredictor to queries
Fix docker build
Update requirements

### CODE REVIEW

File: cogs/commands.py [hunk 1]
```diff
@@ -532,13 +532,19 @@ async def end(self, ctx: discord.ApplicationContext):
...
-        required=True,
-        autocomplete=File_autocompleter.get_indexes,
+        name="user_index",
+        description="Which user file to load the index from",
+        required=False,
+        autocomplete=File_autocompleter.get_user_indexes,
+    )
+    @discord.option(
+        name="server_index",
+        description="Which serever file to load the index from",
```
Comment (Kav-K) [inline cogs/commands.py:542]: typo here

File: models/index_model.py [hunk 2]
```diff
@@ -35,15 +36,14 @@
...
 
-def get_and_query(user_id, index_storage, query, llm_predictor):
-    # TODO Do prediction here for token usage
+def get_and_query(user_id, index_storage, query, response_mode, llm_predictor):
     index: [GPTSimpleVectorIndex, ComposableGraph] = index_storage[
         user_id
     ].get_index_or_throw()
     if isinstance(index, GPTTreeIndex):
-        response = index.query(query, verbose=True, child_branch_factor=2)
+        response = index.query(query, verbose=True, child_branch_factor=2, llm_predictor=llm_predictor)
```
Comment (Kav-K) [inline models/index_model.py:44]: Can you add the corresponding calls to the usage service to update the actual usage whenever we call a `.query` and when we construct GPTTreeIndexes for Deep Compose?

File: cogs/index_service_cog.py [hunk 3]
```diff
@@ -95,8 +95,26 @@ async def discord_backup_command(self, ctx):
...
+            await ctx.respond("Please provide a user or server index")
+            return
+
+        if user_index and server_index:
+            await ctx.respond(
+                "Please provide only one user index or server index. Only one or the other."
+            )
+            return
+
+        if server_index:
```
Comment (Kav-K) [inline cogs/index_service_cog.py:111]: can u update readme slightly with the differences between user and server indexes

**Review by Kav-K — CHANGES_REQUESTED** [review]

File: models/index_model.py (hunk 2 above)
Comment (Hikari-Haru) [reply models/index_model.py:44]: Already counts when the query command is called, added counting to deep compose too

**Pull request merged**